### DISCUSSION TITLE
使用os.exec执行命令打印和直接执行打印不同，空格都没了

### DISCUSSION THREAD

**Discussion by PeakRacing** [xmake-io/xmake#4311] [Q&A]
使用os.exec执行命令打印和直接执行打印不同
os.exec：
![image](https://github.com/xmake-io/xmake/assets/24759100/c48efd76-4bee-4d0a-8350-dfbbd8c272dc)
直接执行：
![image](https://github.com/xmake-io/xmake/assets/24759100/26f3da4a-3e02-4175-b3e0-ee3bdc586b8e)

**Comment by waruqi** [comment]
不清楚，你可以看下这个子进程的源码，看下它是否会根据一些环境配置，做不同的输出

**Reply by PeakRacing** [reply]
这个是闭源的看不到，只能直接调用

**Reply by waruqi** [reply]
这个应该跟 os.exec 没啥关系，应该还是环境信息了 影响了子进程对输出的控制，这个你只能自己调下。os.exec 并不会重定向捕获输出，也不会去修改stdout 的行为，子进程输出什么就是什么

**Reply by PeakRacing** [reply]
这是一个闭源发布的exe，我在cmd 和 powerell测试正常，没找到其他影响地方，有些奇怪 https://gitee.com/openLuat/luatos-soc-2023/blob/master/PLAT/tools/fcelf.exe
